main/EventSystem.cpp

```diff
@@ -3170,12 +3170,13 @@ bool CEventSystem::processLuaCommand(lua_State *lua_state, const std::string &fi
 		if (aParam.size() > 3) {
 			sound = aParam[3];
 		}
-		if (aParam.size() > 4) {
-			if (aParam[4].find("midx_") != std::string::npos) {
-				extraData = aParam[4];
+		if (aParam.size() > 4 && !aParam[4].empty()) {
+			if (isInt(aParam[4]) && aParam.size() > 5 && !aParam[5].empty() && aParam[5] == "fcm") {
```

Comment:
I am personally not happy with this if/else, now you can't combine extraData parts.
For example if you want to send id FROM device 123 TO only mobile divice midx 456.
Then users can fill in midx_456|123?
Or 123|midx_456?
The pipe is only a seperator.

Reply:
Agreed. More refactoring could be useful.

Current 'issue' is that the `extraData` is used for multiple purposes in multiple notification subsystems.

midx for selecting certain receiving mobile devices is only available for FCM and currently only used AFAIK for the 'test' button in the DZ UI.

The specific '|Device=<somedzidx>' is used at more subsystems and now also by FCM.

Before, there was no way to pass a DZ deviceidx to the notification subsystems via the dzVents `notify()`. FCM now (ab)uses this existing method to get the 'missing' device Idx.

Rethinking the setup and goals might be needed and might need more rework 😁